src/bluehawk/actions/copy.ts

```diff
@@ -43,7 +68,15 @@ export const copy = async (
       output,
       path.relative(projectDirectory, path.dirname(filePath))
     );
-    const targetPath = path.join(directory, path.basename(filePath));
+
+    let targetPath = path.join(directory, path.basename(filePath));
+    // rename binary file if a new name is specified
+    {
+      const renamed = getRename(path.basename(filePath));
+      if (renamed) {
+        targetPath = path.join(directory, renamed);
```

Comment:
Minor: It's being done twice in this function. You could refactor this block to be another local function like getRename but it gets the target with the rename applied. Maybe like `getRenameAwareTargetPath(directory, name)`?

Reply:
Great refactor suggestion! much cleaner